### ISSUE TITLE
Auto-complete for Python will not work for installed packages

### ISSUE THREAD

**Issue by Phil-T1** [nvim-lua/kickstart.nvim#222]
Hi,

I'm trying to get auto-complete working for Python packages.

If I import a built-in library such as time, then autocomplete will works great.

But, if I import an installed library such as Numpy or Pandas then auto-complete does not work - the following yields no matches:
import pandas as pd
pd.re

I've tried multiple lsp servers and it's the same every time.

I'm not using a virtual environment for my code.

Does anybody have any idea what I'm doing wrong? I'd ideally like to get the auto-complete for any packages I install.

Thanks very much!

Phil.

**Comment by feoh** [comment]
Hi Phil!

One of the reasons virtual environments help a lot is that they provide a very constrained context for your code and its dependencies.

You say you're not using virtual environments. Are you *sure* the modules you think you have installed are actually installed in the Python runtime you're using?

Also, more details would be helpful here as well. What Python language server are you using? Neovim has several. I use pyright but some people don't like that MSFT makes that so there are other options.

My typical Python setup is as follows:
* Install neovim 0.8 or later
* Ensure everything's up to date with :LazyUpdate or :Lazy and u from the menu.
* Run `:MasonInstall pyright`
* Run `pip install pyright` in the Python my code runs in (I use virtualenvs so mine goes there)

Hope this helps and maybe if you give us more details about the workflow we can help further!

**Comment by Phil-T1** [comment]
Hi Chris,

Thanks very much for responding :)

I'm not using venvs as my use case is basically for docker containers that are already dedicated to single purposes, I just want a terminal editor to SSH in with, so I'm looking to make a standard setup that each container can have.

The docker container is just the latest Ubuntu.

I'm using Python 3.10.6, python-lsp-server, but I've also tried Jedi and get the same issue. I'll give pyright a go, but it requires npm, so I'd have to sort that first. I suspect that isn't the issue but we'll see...

I installed the latest neovim:
https://github.com/neovim/neovim/releases/latest/download/nvim.appimage

I created the init.lua and added the kickstarter text.

I've installed:
python3-venv
neovim module

I've added to the init.lua:
vim.g.python3_host_prog = '/usr/bin/python3.10'
vim.g.python_host_prog = '/usr/bin/python3.10'

Pandas is definitely available, if I use ":python3 import pandas" then it does it. I also installed Pandas on both python locations in the system, just to be sure.

Packages like time and os will auto-complete without issue, but installed ones like pandas will not.
<img width="384" alt="image" src="https://user-images.githubusercontent.com/88270060/226119307-30526af7-1275-4d56-9af1-fda69d20e257.png">
<img width="264" alt="image" src="https://user-images.githubusercontent.com/88270060/226119329-8d69c5c1-642e-4e34-b530-317fd2c4355b.png">

Am I missing something? I really like the idea of a lighter weight vscode through SSH, but I want the auto-complete for all packages!


Thanks very much for your time,

Phil.

**Comment by krlenell** [comment]
Tangential to this, I'm wondering where to set `vim.g.python3_host_prog = 'path/to/python'` in the `init.lua`.

When I run `:checkhealth` I am always told `  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.`.This is not a problem outside of virtual environments. 

I followed [this guide](https://github.com/deoplete-plugins/deoplete-jedi/wiki/Setting-up-Python-for-Neovim#using-virtual-environments), and at worst it should find a faulty path to python, not state that the path to python is missing on a health check. 

I've tried putting it in various places in the `init.lua` with no success.
